tests/Doctrine/Tests/ORM/Functional/Ticket/GH6884Test.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use Doctrine\ORM\Event\PreUpdateEventArgs;
+use Doctrine\ORM\Events;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @group GH-6884
+ */
+final class GH6884Test extends OrmFunctionalTestCase
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->_schemaTool->createSchema(
+            [
+                $this->_em->getClassMetadata(GH6884Person::class)
+            ]
+        );
+
+        $this->_em->getEventManager()->addEventListener(
```

Comment:
> Verify that firing the PreUpdate event succeeds when a flush is called

How are you ensuring that this is called? Maybe using a partial mock and checking the method gets called is a better strategy? Like:

```php
$listener = $this->createPartialMock(\stdClass::class, ['preUpdate']);
$listener->expects($this->once())->method('preUpdate');

$this->_em->getEventManager()->addEventListener([Events::preUpdate], $listener);

Reply:
Yea that is a much better way to perform the assertion. I wasn't aware of the createPartialMock method. I'll refactor to do that